zebra-chain/src/transaction.rs

```diff
@@ -587,11 +595,15 @@ impl Transaction {
                 sapling_shielded_data,
                 ..
             } => {
-                let transparent = transparent_value_balance(inputs, outputs, utxos);
-                let sprout = sprout_value_balance(joinsplit_data);
-                let sapling = sapling_value_balance(sapling_shielded_data);
-
-                ValueBalance::new(transparent, sprout, sapling, None)
+                let transparent = transparent_value_pool(inputs, outputs, utxos);
+                let sprout = sprout_value_pool(joinsplit_data);
+                let sapling = sapling_value_pool(sapling_shielded_data);
+
+                let mut value_balance = ValueBalance::default();
+                value_balance.transparent_value_balance(transparent?);
+                value_balance.sprout_value_balance(sprout?);
+                value_balance.sapling_value_balance(sapling?);
+                Ok(value_balance)
```

Comment:
Idea: I'm wondering if it would help to make the helper functions return `Amount` instead of `ValueBalance`, so that this code could use a constructor like:

```suggestion
                Ok(ValueBalance::new(
                    transparent_amount(inputs, outputs, utxos)?,
                    sprout_amount(joinsplit_data)?,
                    sapling_amount(sapling_shielded_data)?,
                    None,
                ))
```

Going a bit further, we could leverage some version independent `Transaction` getters to make the method more concise. I'm not sure if there are disadvantages to this idea though :thinking: 

```
    pub fn value_balance(
        &self,
        utxos: &HashMap<transparent::OutPoint, transparent::Utxo>,
    ) -> Result<ValueBalance<NegativeAllowed>, Box<dyn std::error::Error>> {
        Ok(ValueBalance::new(
            self.transparent_amount(utxos),
            self.sprout_amount(),
            self.sapling_amount(),
            self.orchard_amount(),
        ))
    }
```

where `Transaction::<pool>_amount` are the equivalent of `<pool>_amount` functions, but with calls to data getters like `self.inputs()`, `self.outputs()`, `self.sprout_joinsplit_data()`, `self.sapling_shielded_data()`, `self.orchard_shielded_data()`.

This sounds a bit complex, so I'm okay if we leave this for later and just open an issue.